File: benchmarks/bench_matmul.go
// bench_matmul.go — AI: Matrix Multiply
// Compile: go build -o bench_matmul_go.exe bench_matmul.go
package main

import ("fmt"; "os")

func matmulSum(n int) int64 {
	var total int64
	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			var acc int64
			for k := 0; k < n; k++ {
				a := int64((i+k)%17)
				b := int64((k*j+1)%13)
				acc += a * b
			}
			total += acc
		}
	}
	return total
}

func main() {
	var result int64
	for rep := 0; rep < 5000; rep++ {
		result = matmulSum(32)
	}
	fmt.Println(result)
	os.Exit(0)
}

File: benchmarks/bench_matmul.js
const { performance } = require('perf_hooks');
function matmulSum(n) {
    let total = 0;
    for (let i = 0; i < n; i++)
        for (let j = 0; j < n; j++) {
            let acc = 0;
            for (let k = 0; k < n; k++)
                acc += ((i + k) % 17) * ((k * j + 1) % 13);
            total += acc;
        }
    return total;
}
const t0 = performance.now();
let result = 0;
for (let rep = 0; rep < 5000; rep++) result = matmulSum(32);
const ms = performance.now() - t0;
console.log(result);
process.stderr.write(`COMPUTE_MS=${ms.toFixed(3)}\n`);
